diff --git a/node_modules/react-native-worklets/android/build.gradle b/node_modules/react-native-worklets/android/build.gradle
index 021f5bc..60f6261 100644
--- a/node_modules/react-native-worklets/android/build.gradle
+++ b/node_modules/react-native-worklets/android/build.gradle
@@ -27,10 +27,11 @@ def resolveReactNativeDirectory() {
     }
 
     // Fallback to node resolver for custom directory structures like monorepos.
+    def nodeExecutable = System.getProperty("NODE_BINARY", "node")
     def reactNativePackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native/package.json')")
+            commandLine(nodeExecutable, "--print", "require.resolve('react-native/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativePackage.exists()) {
@@ -151,7 +152,7 @@ apply plugin: "de.undercouch.download"
 
 
 android {
-    compileSdkVersion safeExtGet("compileSdkVersion", 34)
+    compileSdkVersion safeExtGet("compileSdkVersion", 35)
 
     namespace "com.swmansion.worklets"
 
@@ -176,7 +177,7 @@ android {
 
     defaultConfig {
         minSdkVersion safeExtGet("minSdkVersion", 23)
-        targetSdkVersion safeExtGet("targetSdkVersion", 34)
+        targetSdkVersion safeExtGet("targetSdkVersion", 35)
         versionCode 1
         versionName WORKLETS_VERSION
 
@@ -294,7 +295,7 @@ android {
 
 def validateReactNativeVersionResult = providers.exec {
     workingDir(projectDir.path)
-    commandLine("node", "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
+    commandLine(System.getProperty("NODE_BINARY", "node"), "./../scripts/validate-react-native-version.js", REACT_NATIVE_VERSION.toString())
     ignoreExitValue = true
 }
 
diff --git a/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug
new file mode 100644
index 0000000..c277d55
--- /dev/null
+++ b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/debug
@@ -0,0 +1,18 @@
+{
+  "installationFolder": "/Users/rontoledo/Projects/JeepTrip/JeepTripApp/node_modules/react-native-worklets/android/build/intermediates/prefab_package/debug/prefab",
+  "gradlePath": ":react-native-worklets",
+  "packageInfo": {
+    "packageName": "react-native-worklets",
+    "packageVersion": "0.5.1",
+    "packageSchemaVersion": 2,
+    "packageDependencies": [],
+    "modules": [
+      {
+        "moduleName": "worklets",
+        "moduleHeaders": "/Users/rontoledo/Projects/JeepTrip/JeepTripApp/node_modules/react-native-worklets/android/build/prefab-headers/worklets",
+        "moduleExportLibraries": [],
+        "abis": []
+      }
+    ]
+  }
+}
\ No newline at end of file
diff --git a/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release
new file mode 100644
index 0000000..bd0392b
--- /dev/null
+++ b/node_modules/react-native-worklets/android/build/intermediates/prefab_package_header_only/prefab_publication.json/release
@@ -0,0 +1,18 @@
+{
+  "installationFolder": "/Users/rontoledo/Projects/JeepTrip/JeepTripApp/node_modules/react-native-worklets/android/build/intermediates/prefab_package/release/prefab",
+  "gradlePath": ":react-native-worklets",
+  "packageInfo": {
+    "packageName": "react-native-worklets",
+    "packageVersion": "0.5.1",
+    "packageSchemaVersion": 2,
+    "packageDependencies": [],
+    "modules": [
+      {
+        "moduleName": "worklets",
+        "moduleHeaders": "/Users/rontoledo/Projects/JeepTrip/JeepTripApp/node_modules/react-native-worklets/android/build/prefab-headers/worklets",
+        "moduleExportLibraries": [],
+        "abis": []
+      }
+    ]
+  }
+}
\ No newline at end of file
